bench-vortex/benches/compressor_throughput.rs

```diff
@@ -0,0 +1,125 @@
+use std::collections::HashSet;
+
+use criterion::{black_box, criterion_group, criterion_main, Criterion, Throughput};
+use itertools::Itertools as _;
+use mimalloc::MiMalloc;
+use vortex::array::PrimitiveArray;
+use vortex::compute::unary::try_cast;
+use vortex::validity::Validity;
+use vortex::{IntoArray as _, IntoCanonical};
+use vortex_dtype::PType;
+use vortex_sampling_compressor::compressors::alp::ALPCompressor;
+use vortex_sampling_compressor::compressors::alp_rd::ALPRDCompressor;
+use vortex_sampling_compressor::compressors::bitpacked::{
+    BITPACK_NO_PATCHES, BITPACK_WITH_PATCHES,
+};
+use vortex_sampling_compressor::compressors::delta::DeltaCompressor;
+use vortex_sampling_compressor::compressors::dict::DictCompressor;
+use vortex_sampling_compressor::compressors::r#for::FoRCompressor;
+use vortex_sampling_compressor::compressors::roaring_int::RoaringIntCompressor;
+use vortex_sampling_compressor::compressors::runend::DEFAULT_RUN_END_COMPRESSOR;
+use vortex_sampling_compressor::compressors::zigzag::ZigZagCompressor;
+use vortex_sampling_compressor::compressors::CompressorRef;
+use vortex_sampling_compressor::SamplingCompressor;
+
+#[global_allocator]
+static GLOBAL: MiMalloc = MiMalloc;
+
+fn primitive(c: &mut Criterion) {
+    let mut group = c.benchmark_group("primitive-decompression");
+    let num_values = u16::MAX as u64;
+    group.throughput(Throughput::Bytes(num_values * 4));
+
+    let int_array = PrimitiveArray::from_vec(
+        (0..num_values).map(|i| i as u32 + 100).collect_vec(),
```

Comment:
Is the idea that the numbers here are supposed to be linear? Maybe sample from a distribution or just use random number generator